rfd/0167-console-service.md

```diff
@@ -0,0 +1,225 @@
+---
+authors: Gabriel Corado (gabriel.oliveira@goteleport.com)
+state: draft
+---
+
+# RFD 0167 - Debug service
+
+## Required Approvals
+
+* Engineering: @r0mant && (@greedy52 || @rosstimothy)
+* Security: @reedloden || @jentfoo
+
+## What
+
+This RFD proposes a mechanism for Teleport administrators to switch log-level
+and enable profile consumption without restarting the instance.
+
+## Why
+
+During incidents in production environments, debug logs and diagnostic profiles
+(such as CPU and Memory allocation) are often unavailable for timely
+troubleshooting. To enable those, administrators need to restart the instance,
+potentially disrupting the application state and obscuring the issue they're
+trying to solve.
+
+In addition, the addition will make collecting profiles with a single command
+even on distroless environments where `curl` is not present.
+
+## Details
+
+A separate service will be introduced specifically for troubleshooting
+purposes. This will ensure the current diagnosis service is kept as is, not
+breaking current integrations or introducing behavior changes. The new service
+will also always be available (contrasting with the diagnosis service, which
+is optional and disabled by default), so in scenarios where users need to use
```

Comment:
It seems like there should be an option to disable this feature.  Compliance or security requirements may push a customer to need to disable this.